### PR TITLE
Add runtime logging feature

### PR DESCRIPTION
**Pull request by michimune**
This PR enables the runtime logging feature to the runtime host backend. Currently it doesn't emit ETW events yet, which will be done soon.

### CODE REVIEW

File: src/Diagnostics.RuntimeHost/Utilities/RuntimeLogger.cs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+            _runtimeLogs.AddOrUpdate(reqid,
+                id => new List<RuntimeEventLogEntry> { info }, // Adding
+                (id, logs) => // Updating
+                {
+                    logs.Add(info);
+                    return logs;
+                });
+
+            // TODO: emit ETW event
+            ////DiagnosticsETWProvider.Instance.LogRuntimeHostMessage(info.Message);
```
Comment (rekhaswaminathan) [inline src/Diagnostics.RuntimeHost/Utilities/RuntimeLogger.cs]: Should this comment be removed?

Comment (michimune) [reply src/Diagnostics.RuntimeHost/Utilities/RuntimeLogger.cs]: You are right. I will just leave the TODO comment and delete the next time.

Comment (michimune) [reply src/Diagnostics.RuntimeHost/Utilities/RuntimeLogger.cs]: Removed

File: src/Diagnostics.ModelsAndUtils/Models/QueryResponse.cs [hunk 2]
```diff
@@ -1,8 +1,11 @@
-﻿namespace Diagnostics.ModelsAndUtils.Models
+﻿using System.Collections.Generic;
+
+namespace Diagnostics.ModelsAndUtils.Models
 {
     public class QueryResponse<T>
     {
         public CompilerResponse CompilationOutput;
+        public IEnumerable<RuntimeLogEntry> RuntimeOutput;
```
Comment (ShekharGupta1988) [inline src/Diagnostics.ModelsAndUtils/Models/QueryResponse.cs]: nit: probably rename this to be RuntimeLogOutput

File: src/Diagnostics.RuntimeHost/Diagnostics.RuntimeHost.csproj [hunk 3]
```diff
@@ -48,4 +48,8 @@
     </Content>
   </ItemGroup>
 
+  <Target Name="PreBuild" BeforeTargets="PreBuildEvent">
+    <Exec Command="taskkill /f /fi &quot;WINDOWTITLE eq Diagnostics.RuntimeHost&quot;" />
```
Comment (hforeste) [inline src/Diagnostics.RuntimeHost/Diagnostics.RuntimeHost.csproj]: Its not a big deal but when I run the project, it just somehow reuses the same process so not sure why we might need this.

File: src/Diagnostics.ModelsAndUtils/Models/QueryResponse.cs (hunk 2 above)
Comment (michimune) [reply src/Diagnostics.ModelsAndUtils/Models/QueryResponse.cs]: Renamed.

File: src/Diagnostics.RuntimeHost/Diagnostics.RuntimeHost.csproj (hunk 3 above)
Comment (michimune) [reply src/Diagnostics.RuntimeHost/Diagnostics.RuntimeHost.csproj]: I have reverted this change.

**Pull request merged**